### PR TITLE
[google_sign_in] Clean up pre-Pigeon code

### PR DESCRIPTION
**Pull request by stuartmorgan**
Internal native helper code was still using `id` arguments, doing type checking, and handling `NSNull`, none of which is relevant now that the calling code has been converted to Pigeon and is already strongly typed as `NSString`.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
test-exempt: code refactor with no semantic change

**Comment by stuartmorgan** [comment]
I just happened to see this and it made me sad, so I thought I'd clean it up :)

File: packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m [hunk 1]
```diff
@@ -270,30 +270,19 @@ - (void)addScopes:(NSArray<NSString *> *)scopes
...
-    clientID = clientIDArg;
-  } else if (self.googleServiceProperties) {
-    clientID = self.googleServiceProperties[kClientIdKey];
-  } else {
-    // We couldn't resolve a clientId, without which we cannot create a GIDConfiguration.
+/// @return @c nil if GoogleService-Info.plist not found and runtimeClientIdentifier is not
+/// provided.
+- (GIDConfiguration *)configurationWithClientIdentifier:(NSString *)runtimeClientIdentifier
+                                 serverClientIdentifier:(NSString *)runtimeServerClientIdentifier
+                                           hostedDomain:(id)hostedDomain {
```
Comment (vashworth) [inline packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m:277]: Why is hostedDomain still using id?

File: packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m [hunk 2]
```diff
@@ -270,30 +270,19 @@ - (void)addScopes:(NSArray<NSString *> *)scopes
...
+  NSString *clientID = runtimeClientIdentifier ?: self.googleServiceProperties[kClientIdKey];
```
Comment (vashworth) [inline packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m:278]: Don't you need to check if `self.googleServiceProperties` is null before getting a key from it? Wouldn't it error/crash if it's null?

File: packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m [hunk 3]
```diff
@@ -270,30 +270,19 @@ - (void)addScopes:(NSArray<NSString *> *)scopes
...
 
-  BOOL hasDynamicServerClientId = [serverClientIDArg isKindOfClass:[NSString class]];
-  NSString *serverClientID = hasDynamicServerClientId
-                                 ? serverClientIDArg
-                                 : self.googleServiceProperties[kServerClientIdKey];
-
-  NSString *hostedDomain = nil;
-  if (hostedDomainArg != [NSNull null]) {
-    hostedDomain = hostedDomainArg;
-  }
```
Comment (vashworth) [inline packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m:296]: If hostedDomain is still using `id`, does it still need to do this null check?

File: packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m (hunk 1 above)
Comment (stuartmorgan) [reply packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m]: No idea, I thought I changed all three. Fixed!

File: packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m (hunk 2 above)
Comment (stuartmorgan) [reply packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m:278]: Nope! `foo[bar]` is just a shorthand for `[foo objectForKeyedSubscript:bar]`, so standard messaging rules (a message to `nil` returns zero'd memory, `nil` in this case) apply.

Comment (stuartmorgan) [reply packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m:278]: (Notice that the previous serverClientID code was actually already relying on that, so this code wasn't even consistent about the overkill 🙂)

**Review by vashworth — APPROVED** [review]
LGTM!

**Pull request merged**